### PR TITLE
docs: docker file considerations

### PR DESCRIPTION
**Pull request by emonddr**
## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/Application-generator.md [hunk 1]
```diff
@@ -103,3 +103,65 @@ greeted with a message.
 Once the application has been created, additional generators such as
 [controller generator](Controller-generator.md) can be run from the
 application's root directory to further scaffold the application.
+
+
```
Comment (bajtos) [inline docs/site/Application-generator.md]: I think we use only one empty line, not two?

File: docs/site/Application-generator.md [hunk 2]
```diff
@@ -103,3 +103,65 @@ greeted with a message.
...
+### Dockerfile Considerations
```
Comment (bajtos) [inline docs/site/Application-generator.md]: Can you please add a link to this section to the top, where we are describing `--docker` option? Something along the following lines:

```markdown
`--docker`: Generate Dockerfile and add npm scripts to build/run the project in a docker container. 
See also [Dockerfile considerations](#dockerfile-consideration) below.

File: docs/site/Application-generator.md [hunk 3]
```diff
@@ -103,3 +103,65 @@ greeted with a message.
...
+# Check out https://hub.docker.com/_/node to select a new base image
+FROM node:10-slim
+```
+
+As the comment suggests, the Docker image name is only a default, and you are encouraged to select an appropriate base image for your application.
+
+If your application has a dependency on a node module that requires and invokes build tools during `docker build` (for example : [loopback-connector-db2](https://github.com/strongloop/loopback-connector-db2) ), an error may occur due to missing libraries.
+
+```
+...
```
Comment (bajtos) [inline docs/site/Application-generator.md]: This error message is very long and hides the actual advice to use `node:10` image. Can you please find a way how to move the advice above the error message? For example:

```markdown
If your application has a dependency on a node module that requires and invokes build tools 
during `docker build` (for example : [loopback-connector-db2](https://github.com/strongloop/loopback-connector-db2) ), 
an error may occur due to missing libraries.

Using the Docker image `node:10`, however, will allow the `docker build` step to complete successfully.

Example compilation error:

[here goes the long text]
```

**Comment by raymondfeng** [comment]
@emonddr Please run `npm run preitter:fix` to fix the formatting issue - https://travis-ci.com/strongloop/loopback-next/jobs/239816733.

**Comment by raymondfeng** [comment]
Let's add a link to https://github.com/nodejs/docker-node/blob/master/docs/BestPractices.md#node-gyp-alpine.

File: docs/site/Application-generator.md [hunk 4]
```diff
@@ -103,3 +104,78 @@ greeted with a message.
...
+gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/isexe/mode.js:8:5
+gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:152:21)
+gyp ERR! System Linux 4.9.184-linuxkit
+gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "build" "--IS_DOWNLOADED=true" "--IBM_DB_HOME=/home/node/app/node_modules/ibm_db/installer/clidriver"
+gyp ERR! cwd /home/node/app/node_modules/ibm_db
+gyp ERR! node -v v8.16.1
+gyp ERR! node-gyp -v v3.8.0
+gyp ERR! not ok
+
+    at ChildProcess.exithandler (child_process.js:281:12)
```
Comment (raymondfeng) [inline docs/site/Application-generator.md]: Let's remove the lines below and only leave `gyp ERR!` section.

**Pull request merged**